common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+    command=$1
+    if [ ! -f "$command" ]; then
+        log_error "File $command does not exist. It is mandatory."
+    else
+        cat $command
+    fi
+}
+
+get_cached()
+{
+    # Get general system information from cache
+    # Usage:    get_cached file <passwd,group>
+    #           get_cached command <rpm_qa, rpm_Va, rpm_etc_Va, chconfig>
+    #           get_cached filelist <executable, allmyfiles, rpmrhsignedfiles, rpmtrackedfiles>
+    command=$1
+    key=$2
+    case $command in
+        "file")
+            case $key in
+                "passwd")
+                    if [ ! -f "$PREUPG_PASSWD" ]; then
+                        log_error "File $PREUPG_PASSWD does not exist. It is mandatory."
+                    else
+                        cat $PREUPG_PASSWD
+                    fi
+                    ;;
+                "group")
+                    if [ ! -f "$PREUPG_GROUP" ]; then
+                        log_error "File $PREUPG_GROUP does not exist. It is mandatory."
+                    else
+                        cat $PREUPG_GROUP
+                    fi
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'passwd', 'group'."
+                    return 1
+                    ;;
+            esac
+            ;;
+        "command")
+            case $key in
+                "rpm_qa")
+                    _get_cached_command $PREUPG_RPM_QA
+                    ;;
+                "rpm_Va")
+                    _get_cached_command $PREUPG_ALL_CHANGED
+                    ;;
+                "rpm_etc_Va")
+                    _get_cached_command $PREUPG_CONFIGCHANGED
+                    ;;
+                "chconfig")
+                    _get_cached_command $PREUPG_CHKCONFIG
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'rpm_qa', 'rpm_Va', 'rpm_etc_Va', 'chkconfig'".
+                    return 1
+                    ;;
+            esac
+            ;;
+        "filelist")
+            case $key in
+                "allmyfiles")
+                    _get_cached_command $PREUPG_ALLMYFILES
+                    ;;
+                "executable")
+                    _get_cached_command $PREUPG_EXECUTABLES
+                    ;;
+                "rpmtrackedfiles")
+                    _get_cached_command $PREUPG_RPMTRACKEDFILES
+                    ;;
+                "rpmrhsignedfiles")
+                    _get_cached_command $PREUPG_RPM_RHSIGNED
+                    ;;
+                *)
+                    log_error "Unknown $key for $command. Supported are 'executable', 'allmyfiles', 'rpmtrackedfiles', 'rpmrhsignedfiles'."
+                    return 1
+            esac
+            ;;
+        "info")
+            ;;
+        *)
+            log_error "Unknown command $command. Supported are 'file', 'command', filelist', 'info'."
+            return 1
```

Comment:
This is not true.  'info' is not supported.

I don't think it makes sense to even mention it anywhere.  It just does not exist, OK? ;-)